frontend/src/components/transactions/TransactionDetails.tsx

```diff
@@ -36,11 +44,55 @@ export default class extends React.Component<Props, State> {
       })
       .reduce((accumulator, deposit) => accumulator.add(deposit), new BN(0));
     this.setState({ deposit });
+  };
+
+  setAttachedGas = () => {
```

Comment:
Why is this name?

Reply:
you think it is a broken name or you are not clear about the function?
see this https://github.com/nearprotocol/near-explorer/pull/213#discussion_r387437574

Reply:
I don't see why it is called "set" when it clearly only computes a value and returns it.

Reply:
I would probably rename all three `setDeposit`, `setAttachedGas`, `setTotalFee` (btw, not only it sets the total fee...) into `collect*` counterparts, which would accept `actions` as their function parameters, so make them pure functions, and create a separate method, which will call all of the functions and `setState` only once.

Also, we should trigger the calculation process (the proposed method ^) not only from `componentDidMount`, but also from `componentDidUpdate`.